vault/plugin_catalog.go

```diff
@@ -104,6 +97,28 @@ func (c *PluginCatalog) getPluginTypeFromUnknown(ctx context.Context, logger log
 	return consts.PluginTypeUnknown, nil
 }
 
+func isDatabasePlugin(ctx context.Context, plugin *pluginutil.PluginRunner, logger log.Logger) error {
+	merr := &multierror.Error{}
+	// Attempt to run as database V5 plugin
+	v5Client, err := newdbplugin.NewPluginClient(ctx, nil, plugin, log.NewNullLogger(), true)
+	if err == nil {
+		// Close the client and cleanup the plugin process
+		v5Client.Close()
+		return nil
+	}
+	merr = multierror.Append(merr, err)
+
+	v4Client, err := dbplugin.NewPluginClient(ctx, nil, plugin, log.NewNullLogger(), true)
+	if err == nil {
+		// Close the client and cleanup the plugin process
+		v4Client.Close()
+		return nil
+	}
+	merr = multierror.Append(merr, err)
+
+	return merr
```

Comment:
This ensures that we return a `nil` value and not an interface that's `*multierror.Error` with a nil value.

```suggestion
	return merr.ErrorOrNil()
```

Reply:
![https://emojis.slackmojis.com/emojis/images/1529943438/4161/doh.png?1529943438](https://emojis.slackmojis.com/emojis/images/1529943438/4161/doh.png?1529943438)

Fixed

Reply:
Did you end up pushing new changes? I still see this on the latest commit.

Reply:
Just got pushed